mmdet3d/models/model_utils/transformer.py

```diff
@@ -0,0 +1,280 @@
+            nn.Conv1d(num_pos_feats, num_pos_feats, kernel_size=1))
+
+    def forward(self, xyz):
+        xyz = xyz.transpose(1, 2).contiguous()
+        position_embedding = self.position_embedding_head(xyz)
+        return position_embedding
+
+
+@TRANSFORMER_LAYER_SEQUENCE.register_module()
+class GroupFree3DTransformerDecoder(TransformerLayerSequence):
+    """TransformerDeocder in `Group-Free 3D.
+
+    <https://arxiv.org/abs/2104.00678>`_.
+
+    Args:
+        bbox_coder (:obj:`BaseBBoxCoder`): Bbox coder for encoding and
+            decoding boxes.
+        pred_layer_cfg (dict): Config of classfication and regression
+            prediction layers.
+        transformerlayer (list[obj:`mmcv.ConfigDict`] |
+            obj:`mmcv.ConfigDict`): Config of transformerlayer
+            in TransformerCoder. If it is obj:`mmcv.ConfigDict`,
+             it would be repeated `num_layer` times to a
+             list[`mmcv.ConfigDict`]. Default: None.
+        num_layers (int): The number of `TransformerLayer`. Default: None.
+        init_cfg (obj:`mmcv.ConfigDict`): The Config for initialization.
+            Default: None.
+    """
+
+    def __init__(self,
+                 bbox_coder,
+                 pred_layer_cfg=None,
+                 transformerlayers=None,
+                 num_layers=None,
+                 init_cfg=None):
+        super(GroupFree3DTransformerDecoder, self).__init__(
+            transformerlayers=transformerlayers,
+            num_layers=num_layers,
+            init_cfg=init_cfg)
+
+        self.bbox_coder = build_bbox_coder(bbox_coder)
+        self.num_classes = self.num_sizes = self.bbox_coder.num_sizes
+        self.num_dir_bins = self.bbox_coder.num_dir_bins
+
+        # prediction heads
+        self.prediction_heads = nn.ModuleList()
+        for _ in range(self.num_layers):
+            self.prediction_heads.append(
+                BaseConvBboxHead(
+                    **pred_layer_cfg,
+                    num_cls_out_channels=self._get_cls_out_channels(),
+                    num_reg_out_channels=self._get_reg_out_channels()))
+
+        # query proj and key proj
+        self.decoder_query_proj = nn.Conv1d(
+            self.embed_dims, self.embed_dims, kernel_size=1)
+        self.decoder_key_proj = nn.Conv1d(
+            self.embed_dims, self.embed_dims, kernel_size=1)
+
+        # query position embed
+        self.decoder_self_posembeds = nn.ModuleList()
+        for _ in range(self.num_layers):
+            self.decoder_self_posembeds.append(
+                PositionEmbeddingLearned(6, self.embed_dims))
+        # key position embed
+        self.decoder_cross_posembeds = nn.ModuleList()
+        for _ in range(self.num_layers):
+            self.decoder_cross_posembeds.append(
+                PositionEmbeddingLearned(3, self.embed_dims))
+
+    def init_weights(self):
+        # follow the official Group-Free-3D to init parameters
+        for m in self.modules():
```

Comment:
make sure this can initialize all parameters,  please check that all parameters with dim>1 could be initialized in this loop.